src/lib/github-service.ts

```
@@ -0,0 +1,355 @@
+    return this.request<GitHubUser>('/user');
+  }
+
+  async getRepo(owner: string, repo: string): Promise<GitHubRepo> {
+    return this.request<GitHubRepo>(`/repos/${owner}/${repo}`);
+  }
+
+  async parseRepoUrl(url: string): Promise<{ owner: string; repo: string } | null> {
+    try {
+      // Handle various GitHub URL formats
+      const patterns = [
+        /github\.com\/([^\/]+)\/([^\/]+)(?:\/.*)?$/,
```

ESLint

[error] Unnecessary escape character: \/.

(no-useless-escape)